src/algebra/opposites.lean

```diff
@@ -364,3 +364,11 @@ rfl
 lemma units.coe_op_equiv_symm {R} [monoid R] (u : (units R)ᵒᵖ) :
   (units.op_equiv.symm u : Rᵒᵖ) = op (u.unop : R) :=
 rfl
+
+/-- A hom `α →* β` can equivalently be viewed as a hom `αᵒᵖ →* βᵒᵖ`. This is the action
+of the `ᵒᵖ`-functor on morphisms. -/
+@[simps]
+def monoid_hom.op {α β} [monoid α] [monoid β] (f : α →* β) : αᵒᵖ →* βᵒᵖ :=
```

Comment:
While we're here, can we add the ring_hom, mul_equiv, and ring_equiv versions too? They should be easy

Reply:
It would also be great if we could express this as:
```suggestion
def monoid_hom.op {α β} [monoid α] [monoid β] : (α →* β) ≃ (αᵒᵖ →* βᵒᵖ) :=
```